### ISSUE TITLE
Issue using influxdb2_structmap_derive::FromMap to parse struct

### ISSUE THREAD

**Issue by pittengermd** [aprimadi/influxdb2#1]
I was able to reproduce the query example.  But then when I attempted to add the open field, I get a "Cannot parse out map entry".  I suspect this is because the format of the data being returned from the query does not match the layout of the struct. Here is my Cargo.toml and my main.rs
Cargo.toml
```
[package]
name = "influxdb2_query_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
tokio = { version = "1.20.0", features = [ "full" ]}
influxdb2 = "*"
influxdb2-structmap = "*"
influxdb2-structmap-derive ="*"
futures = "*"
```
src/main.rs
```
use futures::prelude::*;
use influxdb2::models::DataPoint;
use influxdb2::models::Query;
use influxdb2::Client;
use influxdb2_structmap::FromMap;
use std::env;
use futures::prelude::*;


#[derive(Debug, Default, influxdb2_structmap_derive::FromMap)]
pub struct StockPrice {
   ticker: String,
   value: f64,
   open: f64,
   time: i64,
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let host = std::env::var("INFLUXDB_HOST").unwrap();
    let org = std::env::var("INFLUXDB_ORG").unwrap();
    let token = std::env::var("INFLUXDB_TOKEN").unwrap();
    let bucket = env::var("INFLUX_BUCKET").unwrap_or("test1".into());

    let client = Client::new(host, org, token);


    println!("HealthCheck: {:#?}", client.health().await?);
      let points: Vec<DataPoint> = vec![ DataPoint::builder("bar")
      .tag("ticker", "AAPL")
      .field("value", 123.46)
      .field("open", 200.0)
      .field("time", 0)
      .build()?, ];
      client.write(&bucket, stream::iter(points)).await?;
      let qs = format!("from(bucket: \"test1\") 
      |> range(start: -1w)
      |> filter(fn: (r) => r.ticker == \"{}\") 
      |> last()
   ", "AAPL");
       let query = Query::new(qs.to_string());

    println!("Query result was: {:#?}", client.query::<StockPrice>(Some(query))
    .await?);

    Ok(())
}
```

Here is the relevant cargo expand output.
```
pub struct StockPrice {
    ticker: String,
    value: f64,
    open: f64,
    time: i64,
}

#[automatically_derived]
#[allow(unused_qualifications)]
impl ::core::default::Default for StockPrice {
    #[inline]
    fn default() -> StockPrice {
        StockPrice {
            ticker: ::core::default::Default::default(),
            value: ::core::default::Default::default(),
            open: ::core::default::Default::default(),
            time: ::core::default::Default::default(),
        }
    }
}
use influxdb2_structmap::value::Value;
use influxdb2_structmap::GenericMap;
impl FromMap for StockPrice {
    fn from_genericmap(mut hashmap: GenericMap) -> StockPrice {
        let mut settings = StockPrice::default();
        let mut key = String::from("ticker");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::String(v) = entry.get() {
                    settings.ticker = v.clone();
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        let mut key = String::from("value");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::Double(v) = entry.get() {
                    settings.value = *v;
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        let mut key = String::from("open");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::Double(v) = entry.get() {
                    settings.open = *v;
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        let mut key = String::from("time");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::Long(v) = entry.get() {
                    settings.time = *v;
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        settings
    }
}
```

**Comment by pittengermd** [comment]
I forgot to add, I did verify with cargo expand that from_genericmap does have keys for all the struct members.  Here are the relevant parts of cargo expand output
```
pub struct StockPrice {
    ticker: String,
    value: f64,
    open: f64,
    time: i64,
}

#[automatically_derived]
#[allow(unused_qualifications)]
impl ::core::default::Default for StockPrice {
    #[inline]
    fn default() -> StockPrice {
        StockPrice {
            ticker: ::core::default::Default::default(),
            value: ::core::default::Default::default(),
            open: ::core::default::Default::default(),
            time: ::core::default::Default::default(),
        }
    }
}
use influxdb2_structmap::value::Value;
use influxdb2_structmap::GenericMap;
impl FromMap for StockPrice {
    fn from_genericmap(mut hashmap: GenericMap) -> StockPrice {
        let mut settings = StockPrice::default();
        let mut key = String::from("ticker");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::String(v) = entry.get() {
                    settings.ticker = v.clone();
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        let mut key = String::from("value");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::Double(v) = entry.get() {
                    settings.value = *v;
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        let mut key = String::from("open");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::Double(v) = entry.get() {
                    settings.open = *v;
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        let mut key = String::from("time");
        if !hashmap.contains_key(&key) {
            key = {
                let res = ::alloc::fmt::format(::core::fmt::Arguments::new_v1(
                    &["_"],
                    &[::core::fmt::ArgumentV1::new_display(&key)],
                ));
                res
            };
        }
        match hashmap.entry(key) {
            ::std::collections::btree_map::Entry::Occupied(entry) => {
                if let Value::Long(v) = entry.get() {
                    settings.time = *v;
                }
            }
            _ => ::std::rt::begin_panic("Cannot parse out map entry"),
        }
        settings
    }
}
```

**Comment by aprimadi** [comment]
@pittengermd 

Hey Matt, 

Currently this library doesn't support multiple fields in the same point. Here is the reason: InfluxDB treats multiple fields as separate data point. 

Here's an example data returned back from InfluxDB in line protocol format:
```
#datatype,string,long,dateTime:RFC3339,dateTime:RFC3339,dateTime:RFC3339,double,string,string,string
#group,false,false,true,true,false,false,true,true,true
#default,_result,,,,,,,,
,result,table,_start,_stop,_time,_value,_field,_measurement,ticker
,,0,2022-07-16T11:01:26.049760474Z,2022-07-23T11:01:26.049760474Z,2022-07-23T11:01:26.046209552Z,200,open,bar,AAPL
,,1,2022-07-16T11:01:26.049760474Z,2022-07-23T11:01:26.049760474Z,2022-07-23T11:01:26.046209552Z,123.46,value,bar,AAPL
```

Or in JSON format so perhaps it's easier to see:
```
{"_field": String("open"), "_measurement": String("bar"), "_start": TimeRFC(2022-07-16T11:01:26.049760474+00:00), "_stop": TimeRFC(2022-07-23T11:01:26.049760474+00:00), "_time": TimeRFC(2022-07-23T11:01:26.046209552+00:00), "_value": Double(200.0), "result": String("_result"), "table": Long(0), "ticker": String("AAPL")}
{"_field": String("value"), "_measurement": String("bar"), "_start": TimeRFC(2022-07-16T11:01:26.049760474+00:00), "_stop": TimeRFC(2022-07-23T11:01:26.049760474+00:00), "_time": TimeRFC(2022-07-23T11:01:26.046209552+00:00), "_value": Double(123.46), "result": String("_result"), "table": Long(1), "ticker": String("AAPL")}
```

I guess there are ways around this to treat multiple data points and combine them into one struct but at the moment I'm kinda busy so not sure when I'm going to add that feature. Of course pull requests are welcome. Take a look at this: https://github.com/aprimadi/influxdb2/blob/main/src/api/query.rs#L92-L99, if we can group multiple data point with the same `_time` field into the same `genericmap` it will probably solve the issue.

A simple workaround for this is to treat the other field as a tag and store it as a string and perform conversion later on.
```rust
#[derive(Debug, Default, influxdb2_structmap_derive::FromMap)]
pub struct StockPrice {
   ticker: String,
   value: f64,
   open: String,
   time: DateTime<FixedOffset>,
}

let points: Vec<DataPoint> = vec![
    DataPoint::builder("bar")
      .tag("ticker", "AAPL")
      .tag("open", "200.0")
      .field("value", 123.46)
      .build()?, 
];
```
The drawback of this approach is that tags are indexed so it's going to use more disk space (for the index files even though the index isn't going to be used).

**Comment by aprimadi** [comment]
This is fixed: 2db0df569305602d115a3a48ae06a1593c5b47f8
